extensions/close-apps/src/manage-whitelist.tsx

```diff
@@ -0,0 +1,211 @@
+        isWhitelisted: whitelistedApps?.includes(app) || app === "Raycast",
+      }));
+
+      return allApps;
+    },
+    [whitelistedApps],
+  );
+
+  const { data: openApps, isLoading: openAppsLoading } = usePromise(
+    async (whitelistedApps: string[] | undefined) => {
+      const openApps = await getOpenApps();
+      return openApps.map((app) => ({
+        name: app,
+        isWhitelisted: whitelistedApps?.includes(app) || app === "Raycast",
+      }));
+    },
+    [whitelistedApps],
+  );
+
+  const apps = useMemo(() => {
+    switch (listState) {
+      case "all":
+        return allApps;
+      case "open":
+        return openApps;
+      case "whitelisted":
+        return Array.from(new Set(whitelistedApps)).map((app) => ({
+          name: app,
+          isWhitelisted: true,
+        }));
+      default:
+        return [];
+    }
+  }, [listState, allApps, openApps, whitelistedApps]);
+
+  const toggleWhitelist = (application: Application) => {
+    const newWhitelist = new Set(
+      application.isWhitelisted
+        ? whitelistedApps?.filter((app) => app !== application.name) || []
+        : [...(whitelistedApps || []), application.name],
+    );
+
+    setWhitelistedApps(Array.from(newWhitelist));
+    revalidate();
+
+    showToast({
+      style: Toast.Style.Success,
+      title: `${application.name} ${application.isWhitelisted ? "removed from" : "added to"} whitelist`,
+    });
+  };
+
+  const closeAllNonWhitelisted = async () => {
+    try {
+      await closeNotWhitelisted();
+      showToast({
+        style: Toast.Style.Success,
+        title: "Closed all non-whitelisted apps",
+      });
+      revalidate();
+    } catch (error) {
+      showToast({
+        style: Toast.Style.Failure,
+        title: "Failed to close apps",
+        message: String(error),
+      });
+    }
+  };
+
+  const getAdjacentListState = (currentState: ListState, type: "next" | "prev") => {
+    const states = Object.keys(StateToTitle) as ListState[];
+    const currentIndex = states.indexOf(currentState);
+    const nextIndex =
+      type === "next" ? (currentIndex + 1) % states.length : (currentIndex - 1 + states.length) % states.length;
+    return states[nextIndex];
+  };
+
+  const cycleListState = () => {
+    const nextState = getAdjacentListState(listState, "next");
+    setListState(nextState);
+  };
+
+  const nextStateTitle = `Show ${StateToTitle[getAdjacentListState(listState, "next")]}`;
+  const prevStateTitle = `Show ${StateToTitle[getAdjacentListState(listState, "prev")]}`;
+
+  const extractIcon = (app: string) => {
+    const applicationPaths = [`/Applications/${app}.app`, `/System/Applications/${app}.app`];
+    return (
+      applicationPaths.find((path) => {
+        try {
+          return fs.existsSync(path);
+        } catch {
+          return false;
+        }
+      }) || `/Applications/${app}.app`
+    );
+  };
+
+  return (
+    <List
+      navigationTitle={StateToTitle[listState]}
```

Comment:
Don't change the `navigationTitle` in the root command - it will be automatically set to the command name. Use `navigationTitle` only in nested screens to provide additional